src/main/java/org/iot/dsa/dslink/java/v2/jdbc/DBConnectionNode.java

```diff
@@ -0,0 +1,346 @@
+    //Methods - Overrides
+    ///////////////////////////////////////////////////////////////////////////
+
+    @Override
+    protected void declareDefaults() {
+        super.declareDefaults();
+        //Default Values
+        declareDefault(JDBCv2Helpers.DB_NAME, DSString.valueOf("No Name"));
+        declareDefault(JDBCv2Helpers.DB_USER, DSString.valueOf("No Name"));
+        declareDefault(JDBCv2Helpers.DB_URL, DSString.valueOf("No URL"));
+        declareDefault(JDBCv2Helpers.DRIVER, DSString.valueOf("No Driver"));
+        declareDefault(JDBCv2Helpers.DB_PASSWORD, DSPasswordAes.valueOf("No Pass")).setHidden(true);
+        declareDefault(JDBCv2Helpers.STATUS, DSString.valueOf(ConnStates.Unknown)).setReadOnly(true);
+        declareDefault(JDBCv2Helpers.LAST_SUCCESS, DSString.valueOf("None")).setReadOnly(true);
+        declareDefault(JDBCv2Helpers.LAST_FAIL, DSString.valueOf("None")).setReadOnly(true);
+        //Default Actions
+        declareDefault(JDBCv2Helpers.QUERY, makeQueryAction());
+        declareDefault(JDBCv2Helpers.EDIT, makeEditAction());
+        declareDefault(JDBCv2Helpers.UPDATE, makeUpdateAction());
+        //TODO: Add streaming Queries
+        //declareDefault(JDBCv2Helpers.STREAM_QUERY, makeStreamingQueryAction());
+        declareDefault(JDBCv2Helpers.REMOVE, makeRemoveDatabaseAction());
+    }
+
+    @Override
+    protected void onStable() {
+        createDatabaseConnection();
+    }
+
+    abstract void createDatabaseConnection();
+
+    ///////////////////////////////////////////////////////////////////////////
+    //Methods - Helpers
+    ///////////////////////////////////////////////////////////////////////////
+
+    void setParameters(DSMap params) {
+        if (!params.isNull(JDBCv2Helpers.DB_NAME))
+            put(db_name, params.get(JDBCv2Helpers.DB_NAME));
+        if (!params.isNull(JDBCv2Helpers.DB_URL))
+            put(db_url, params.get(JDBCv2Helpers.DB_URL));
+        if (!params.isNull(JDBCv2Helpers.DB_USER))
+            put(usr_name, params.get(JDBCv2Helpers.DB_USER));
+        if (!params.isNull(JDBCv2Helpers.DRIVER))
+            put(driver, params.get(JDBCv2Helpers.DRIVER));
+        if (!params.isNull(JDBCv2Helpers.DB_PASSWORD))
+            setCurPass(params.get(JDBCv2Helpers.DB_PASSWORD).toString());
+    }
+
+    String getCurPass() {
+        return ((DSPasswordAes) password.getValue()).decode();
+    }
+
+    private void setCurPass(String pass) {
+        put(password, DSPasswordAes.valueOf(pass));
+    }
+
+    void connSuccess(boolean success) {
+        DSDateTime stamp = DSDateTime.valueOf(System.currentTimeMillis());
+        if (success) {
+            put(conn_status, DSString.valueOf(ConnStates.Connected));
+            put(conn_succ, stamp);
+        } else {
+            put(conn_status, DSString.valueOf(ConnStates.Failed));
+            put(conn_fail, stamp);
+        }
+    }
+
+    void testConnection() {
+        Connection conn = null;
```

Comment:
I don't think there's a point in declaring these outside the try catch block

Reply:
The finally call is impossible otherwise.